pkg/reconciler/kubernetes/tektonchain/tektonchain.go

```diff
@@ -509,3 +515,49 @@ func (r *Reconciler) markUpgrade(ctx context.Context, tc *v1alpha1.TektonChain)
 	}
 	return v1alpha1.RECONCILE_AGAIN_ERR
 }
+
+func AddControllerEnv(controllerEnvs []corev1.EnvVar) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "Deployment" || len(controllerEnvs) == 0 {
+			return nil
+		}
+
+		d := &appsv1.Deployment{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, d)
+		if err != nil {
+			return err
+		}
+
+		for _, v := range controllerEnvs {
```

Comment:
first find the container and iterate the `controllerEnvs` inside the container, it can reduce the number of loops.